tpm2/policy.go

```diff
@@ -235,6 +249,46 @@ func newKeyDataPolicy(alg tpm2.HashAlgorithmId, key *tpm2.Public, pcrPolicyCount
 		}
 	}
 
+	trial := util.ComputeAuthPolicy(alg)
+	trial.PolicyAuthorize(computeV3PcrPolicyRefFromCounterName(pcrPolicyCounterName), keyName)
+	trial.PolicyAuthValue()
+
+	return &keyDataPolicy_v3{
+		StaticData: &staticPolicyData_v3{
+			AuthPublicKey:          key,
+			PCRPolicyCounterHandle: pcrPolicyCounterHandle},
+		PCRData: &pcrPolicyData_v3{
+			PolicySequence:            pcrPolicySequence,
+			AuthorizedPolicySignature: &tpm2.Signature{SigAlg: tpm2.SigSchemeAlgNull}}}, trial.GetDigest(), nil
```

Comment:
all previous code wasn't setting AuthorizedPolicySignature at this stage, any particular reason to do so now?

Reply:
Setting AuthorizedPolicySignature makes it a valid object so that it can be serialized and unserialized, as the zero value for SigAlg is invalid. This is needed here due to the way sealed object creation works with the new sealing API that's built on this.